nselib/stdnse.lua

```diff
@@ -1445,4 +1445,28 @@ function keys(t)
   return ret
 end
 
+-- Returns the case insensitive pattern of given parameter
+-- Useful while doing case insensitive pattern match using string library.
+--
+-- Ex: generate_case_insensitive_pattern("user") = "[uU][sS][eE][rR]"
+--
```

Comment:
Need to include attribution for this function and any others we copy from StackOverflow. https://meta.stackexchange.com/questions/271080/the-mit-license-clarity-on-using-code-on-stack-overflow-and-stack-exchange